### PR TITLE
orm(engine): Orm support context

### LINKED ISSUE #5757: orm(engine): orm client need support context
### Is your feature request related to a problem?

orm client need support context

### Describe the feature you'd like

Add context support for underlying orm layer

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #5757 

### What is changed and how it works?
Add WithContext(ctx) to gorm.DB

#### Tests 

 - Unit test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/pkg/orm/client.go [hunk 1]
```diff
@@ -629,49 +677,52 @@ func (c *metaOpsClient) QueryResourcesByJobID(ctx context.Context, jobID string)
...
+			return nil, cerrors.ErrMetaEntryNotFound.Wrap(err)
 		}
-		return nil, cerrors.ErrMetaOpFail.Wrap(result.Error)
+		return nil, cerrors.ErrMetaOpFail.Wrap(err)
 	}
 	return &ret, nil
 }
 
-// Result defines a query result interface
+// Result defines a query err interface
```
Comment (amyangfei) [inline engine/pkg/orm/client.go:725]: Seems not query error

File: engine/pkg/orm/client_test.go [hunk 2]
```diff
@@ -1243,8 +1242,6 @@ func TestError(t *testing.T) {
 }
 
 func TestLogicEpoch(t *testing.T) {
-	t.Parallel()
-
```
Comment (amyangfei) [inline engine/pkg/orm/client_test.go:1247]: Can parallel be enabled here?

File: engine/pkg/orm/client_test.go [hunk 3]
```diff
@@ -1270,6 +1267,33 @@ func TestLogicEpoch(t *testing.T) {
...
+	defer cancel()
+
+	cli, err := NewMockClient()
+	require.NoError(t, err)
+	defer cli.Close()
+
+	// test normal function
+	err = failpoint.Enable("github.com/pingcap/tiflow/engine/pkg/orm/initializedDelay", "sleep(2000)")
+	require.NoError(t, err)
+	err = cli.Initialize(ctx)
```
Comment (amyangfei) [inline engine/pkg/orm/client_test.go:1281]: We can use `InjectContext` to implement concurrent injection with failpoint, such as https://github.com/amyangfei/concurrent-failpoint

File: engine/pkg/orm/client.go (hunk 1 above)
Comment (maxshuang) [reply engine/pkg/orm/client.go:725]: Wrong replace operation.

File: engine/pkg/orm/client_test.go (hunk 2 above)
Comment (maxshuang) [reply engine/pkg/orm/client_test.go]: Done

File: engine/pkg/orm/client_test.go [hunk 4]
```diff
@@ -1270,6 +1269,46 @@ func TestLogicEpoch(t *testing.T) {
...
+	require.NoError(t, err)
+	defer cli.Close()
+
+	// test normal function
+	err = failpoint.Enable("github.com/pingcap/tiflow/engine/pkg/orm/initializedDelay", "sleep(2000)")
+	require.NoError(t, err)
+	ctx = failpoint.WithHook(ctx, func(ctx context.Context, fpname string) bool {
+		return ctx.Value(fpname) != nil
+	})
+	tx := context.WithValue(ctx, "github.com/pingcap/tiflow/engine/pkg/orm/initializedDelay", struct{}{})
```
Comment (amyangfei) [inline engine/pkg/orm/client_test.go:1288]: `tx` is not readable, better to use ctx2 or newCtx

**Review by amyangfei — COMMENTED** [review]
rest lgtm

File: engine/pkg/orm/client_test.go (hunk 4 above)
Comment (maxshuang) [reply engine/pkg/orm/client_test.go]: Fix.

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**